src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xencoder/HaivisionXEncoderCommunicator.java

```diff
@@ -332,16 +383,26 @@ public void controlProperty(ControllableProperty controllableProperty) {
 		}
 		reentrantLock.lock();
 		try {
-			if (localExtendedStatistics == null) {
+			if (localExtendedStatistics == null || localCreateOutputStream == null) {
 				return;
 			}
 			Map<String, String> extendedStatistics = localExtendedStatistics.getStatistics();
+			Map<String, String> updateCreateOutputStream = localCreateOutputStream.getStatistics();
 			List<AdvancedControllableProperty> advancedControllableProperties = localExtendedStatistics.getControllableProperties();
-			String propertiesAudioAndVideo = property.substring(0, EncoderConstant.AUDIO.length());
-			if (EncoderConstant.AUDIO.equals(propertiesAudioAndVideo)) {
-				controlAudioProperty(property, value, extendedStatistics, advancedControllableProperties);
+			List<AdvancedControllableProperty> advancedControllableCreateOutputProperties = localCreateOutputStream.getControllableProperties();
+
+			if (property.contains(EncoderConstant.CREATE_STREAM)) {
```

Comment:
Maybe property.split here is better, because there are cases for example: A#CreateStream => the group name is not CreateStream but this condition still equals to true.